app/src/main/java/com/rootstrap/android/ui/activity/main/targetpoint/CreateTargetView.kt

```diff
@@ -124,35 +150,100 @@ class CreateTargetView(
         val isTitleValid = targetPointsViewModel.isTitleValid(title)
-        if (isTitleValid.not()) {
-            bindingRoot.title_text_input_layout.error = SHOW_EMPTY_ERROR
-        }
+        bindingRoot.title_text_input_layout.error = if (isTitleValid.not()) {
+            SHOW_EMPTY_ERROR
+        } else null
         return isTitleValid
     }
 
     private fun validateTopic(topic: TopicModel?): Boolean {
         val isTopicValid = targetPointsViewModel.isTopicValid(topic)
-        if (isTopicValid.not()) {
-            bindingRoot.topic_text_input_layout.error = SHOW_EMPTY_ERROR
-        }
+        bindingRoot.topic_text_input_layout.error = if (isTopicValid.not()) {
+            SHOW_EMPTY_ERROR
+        } else null
         return isTopicValid
     }
 
     fun expandCollapseCreateTargetSheet() {
         bottomSheetBehavior.state = if (bottomSheetBehavior.state != BottomSheetBehavior.STATE_EXPANDED) {
+            showBottomButtons(true)
+            eraseTargetInformation()
+            BottomSheetBehavior.STATE_EXPANDED
+        } else {
+            BottomSheetBehavior.STATE_COLLAPSED
+        }
+    }
+
+    private fun showBottomButtons(isCreateTarget: Boolean) {
+        with(bindingRoot) {
+            if (isCreateTarget) {
+                small_buttons_container.visibility = View.GONE
+                save_target_btn.visibility = View.VISIBLE
+            } else {
+                small_buttons_container.visibility = View.VISIBLE
+                save_target_btn.visibility = View.GONE
+            }
+        }
+    }
+
+    private fun eraseTargetInformation() {
+        with(bindingRoot) {
+            title_edit_text.text = Util.createEmptyEditable()
+            area_edit_text.text = Util.createEmptyEditable()
+            topic_edit_text.text = Util.createEmptyEditable()
+            text_select_topic.text = context.getString(R.string.select_a_topic)
+            text_area_length.text = context.getString(R.string.specify_area_length)
+        }
+    }
+
+    private fun showTargetInformation(targetModel: TargetModel) {
+        bottomSheetBehavior.state = if (bottomSheetBehavior.state != BottomSheetBehavior.STATE_EXPANDED) {
+            completeTargetInformation(targetModel)
             BottomSheetBehavior.STATE_EXPANDED
         } else {
             BottomSheetBehavior.STATE_COLLAPSED
         }
     }
 
+    private fun completeTargetInformation(targetModel: TargetModel) {
+        with(bindingRoot) {
+            selectedTarget = targetModel
+            showBottomButtons(false)
+            title_edit_text.text = Util.createEditable(targetModel.title)
+            area_edit_text.text = Util.createEditable("" + targetModel.radius)
+            targetModel.topic?.run {
+                selectedTopic(this)
+            }
+            text_select_topic.text = context.getString(R.string.topic)
+            text_area_length.text = context.getString(R.string.area_length)
```

Comment:
I think that you can do this in the next PR, but you're missing the topic icon, and the distance metric (m),
but like I said, just keep in mind for the next PR.

Reply:
I added the topic icon in the topic selection